### PR TITLE
Move loadModule and paintAsType from Target to Compiler

### PR DESCRIPTION
**Pull request by ibuclaw**
These are a couple of funny little functions that are compiler-specific, but are not directly related to the target platform.

### CODE REVIEW

**Comment by wilzbach** [comment]
(rebased)

File: src/posix.mak [hunk 1]
```diff
@@ -444,7 +444,7 @@ $G/backend.a: $(G_OBJS) $(SRC_MAKE)
 	$(AR) rcs $@ $(G_OBJS)
 
 $G/lexer.a: $(LEXER_SRCS) $(LEXER_ROOT) $(HOST_DMD_PATH) $(SRC_MAKE)
-	CC="$(HOST_CXX)" $(HOST_DMD_RUN) -lib -of$@ $(MODEL_FLAG) -J$G -L-lstdc++ $(DFLAGS) $(LEXER_SRCS) $(LEXER_ROOT)
+	CC="$(HOST_CXX)" $(HOST_DMD_RUN) -lib -of$@ $(MODEL_FLAG) -J$G -J../res -L-lstdc++ $(DFLAGS) $(LEXER_SRCS) $(LEXER_ROOT)
```
Comment (WalterBright) [inline src/posix.mak]: The additional -J switch appears to have nothing to do with this PR.

Comment (ibuclaw) [reply src/posix.mak]: The circular import adds extra module dependencies on the bare bones "lexer" library. I have been making other PRs to reduce the effect of this, but ultimately will probably move the vendor field back to globals for now.

Comment (WalterBright) [reply src/posix.mak]: Ok. The rest of the PR looks good to me.

Comment (jacob-carlborg) [reply src/posix.mak]: Can we remove the lexer library from the makefiles? We already have the Dub package.

**Comment by ibuclaw** [comment]
Rebased.

**Comment by ibuclaw** [comment]
Rebased again.

**Review by PetarKirov — APPROVED** [review]

**Pull request merged**